be/src/storage/lake/update_compaction_state.cpp

```diff
@@ -29,6 +29,8 @@ CompactionState::~CompactionState() {
     if (_update_manager != nullptr) {
         _update_manager->compaction_state_mem_tracker()->release(_memory_usage);
     }
+    // make sure segment iters release before `_stats`
```

Comment:
If you want the `_segment_iters` to be destructed before `_stats`, why not just declare `_stats` before the `_segment_iters`?

Reply:
make sense, done.